Keyboard zoom

eridilla opened the pull request:
Issue #189

rugk commented on src/popup/qrcode.html:
```diff
@@ -22,6 +22,9 @@
 			</div>
 		</div>
+		<div>
+			<p id="zoomTooltip">Alt +/- to zoom</p>
```
That would need to be localized, otherwise, it's static in all languages.

But let's first see whether that is a the appropriate approach.

For more information, please [see the MDN guide on how to localize WebExtensions](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Internationalization#Providing_localized_strings_in__locales) and [our internationalization (i18n) guide](https://github.com/TinyWebEx/common/blob/master/CONTRIBUTING.md#internationalisation-i18n).

rugk commented on src/popup/modules/UserInterface.js:
```diff
@@ -591,72 +629,109 @@ export function init() {
+function zoomQrCode(e) {
+    var evtobj = window.event ? event : e;
+    if (evtobj.keyCode == 61 && evtobj.altKey && qrLastSize < 440) {
+        setNewQrCodeSize(qrLastSize + 30, true);
```
Note `setNewQrCodeSize` may – depending on https://github.com/rugk/offline-qr-code/blob/220d6f3c7394ba4dc5a8d831edd3807838041c9a/src/popup/modules/UserInterface.js#L242 save the size to add-on storage's, too. THough thinking about it, that makes sense, because it is then just persisted.

So yeah, I dunno whether my idea of remembering it additionally, was a good idea, but it seems okay'ish.

rugk commented on src/popup/modules/UserInterface.js:
```diff
@@ -591,60 +629,109 @@ export function init() {
+}
+
+// zoom listener
+document.onkeydown = zoomQrCode;
```
IMHO better use `addEventListener` it's the modern and better syntax.

rugk commented on src/popup/modules/UserInterface.js:
```diff
@@ -591,60 +629,109 @@ export function init() {
+
+        localStorage.setItem("lastSize", qrLastSize);
+    }
+    if (evtobj.keyCode == 173 && evtobj.altKey && qrLastSize > 50) {
```
`keyCode` is kinda deprecated, see the warning at https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode, so let's better switch to alternatives, if possible.

rugk reviewed (changes requested):
Hi @eridilla,
thanks for your first contribution to this project! :tada: :+1: 🏅
I hope you'll like this project and enjoy hacking on it… :smiley:

I am a little critical with having such a non-standard tooltip like <kbd>Alt</kbd>+ (+/-) for zoomimng, given that as I said in https://github.com/rugk/offline-qr-code/issues/189#issuecomment-1284404490 mouse wheel and zoom also already works.

It's a good idea to give a hint, but maybe that is a little too obtrusive (at best it should work intuitively, but I see why that was not possible, as you've explained in https://github.com/rugk/offline-qr-code/issues/189#issuecomment-1356860550).

So the approach I would recommend, as that is not the most important feature here, would be to add a tip via the (already integrated) [RandomTips library](https://github.com/TinyWebEx/RandomTips/).

Also I need to test this out, of course.

---

A general note, please: Next time avoid reformatting many unrelated files, that are not related to the change. The format looks okay, so I will keep that (dunno how that happened? EsLint?), but it makes reviewing harder. Or, at least, do it in a separate commit.

Also, BTW: Next time, try to avoid doing a pull request from the master branch, because you can run into problems when you have a "non-clean" master that does not follow this repo here (i.e. "upstream"). [See this article for details](https://help.github.com/en/articles/proposing-changes-to-your-work-with-pull-requests). Anyway, this is only a tip for the next time. :smiley:

eridilla commented:
Hey @rugk,
thank you for the review and the tips!

I looked at your comments and fixed accordingly. I rewrote the key press listener to the addEventListener function and replace the deprecated keyCodes. I removed the div with the tooltip and added the tip to Tips.js and the english messages.json.

When I was trying out the random tip I got it to show up incorrectly a few times, then changed my code to look like the other tips and then I couldn't get any tip to show at all so I hope the additions I made are correct.

Thank you once again for your comments and I hope these changes are better :)

rugk commented on src/popup/modules/UserInterface.js:
```diff
@@ -712,26 +712,19 @@ export function init() {
-function zoomQrCode(e) {
-    var evtobj = window.event ? event : e;
-    if (evtobj.keyCode == 61 && evtobj.altKey && qrLastSize < 440) {
+document.addEventListener('keydown', function(event) {
```
naaaw, I would still keep the function `zoomQrCode` or so you had… You can just reference that here AFAIK.

rugk reviewed (approved):
Thanks a lot, yeah that looks good now.

Do you want to add yourself to the Contributors file?

rugk commented on src/popup/modules/UserInterface.js:
```diff
@@ -591,60 +629,102 @@ export function init() {
+        localStorage.setItem("lastSize", qrLastSize);
+    }
+    if (event.key == '-' && event.altKey && qrLastSize > 50) {
+        setNewQrCodeSize(qrLastSize - 30, true);
```
Ah and could you extract the `30` into a `const KEYBOARD_ZOOM_SIZE = 30` constant or so? Magic numbers are not a good practise in source code that should be readable… :upside_down_face:

rugk replied:
Same for `50` which I assume is something as `MINIMAL_ZOOM_SIZE` or so?

Just place the constants at the top, there should be some, already. :wink:
